addons/l10n_in_ewaybill_irn/models/l10n_in_ewaybill.py

```diff
@@ -43,11 +43,18 @@ def _prepare_ewaybill_transportation_json_payload(self):
         return super()._prepare_ewaybill_transportation_json_payload()
 
     def _ewaybill_generate_irn_json(self):
-        return {
+        json_payload = {
             'Irn': self.account_move_id._get_l10n_in_edi_response_json().get('Irn'),
             'Distance': str(self.distance),
             **self._prepare_ewaybill_transportation_json_payload(),
+            'DispDtls': self.env['account.edi.format']._get_l10n_in_edi_partner_details(
+                self.partner_ship_from_id, set_phone_and_email=False, set_vat=False)
         }
+        if self._get_billing_partner().l10n_in_gst_treatment == 'overseas':
```

Comment:
if it's base on IRN then `account_move_id` always there so first we consider what is set in document.
```suggestion
        if self.account_move_id.l10n_in_gst_treatment == 'overseas':
```